File: How to transform XML using XSLT, all parameters as strings. No file system/[C#]-How to transform XML using XSLT, all parameters as strings. No file system/C#/XmlTransformation/Core/XmlSerializerHelper.cs
namespace XmlTransformation
{
	using System.IO;
	using System.Xml;
	using System.Xml.Serialization;

	public class XmlSerializerHelper
	{
		public string Serialize(object obj)
		{
			var serializer = new XmlSerializer(obj.GetType());

			var writerSettings = 
				new XmlWriterSettings
					{
						OmitXmlDeclaration = true,
						Indent = true
					};

			var emptyNameSpace = new XmlSerializerNamespaces();
			emptyNameSpace.Add(string.Empty, string.Empty);

			var stringWriter = new StringWriter();
			using (var xmlWriter = XmlWriter.Create(stringWriter, writerSettings))
			{
				serializer.Serialize(xmlWriter, obj, emptyNameSpace);

				return stringWriter.ToString();
			}
		}
	}
}

File: How to transform XML using XSLT, all parameters as strings. No file system/[VB]-How to transform XML using XSLT, all parameters as strings. No file system/VB/XmlTransformation/Core/XmlSerializerHelper.vb
Imports System.Xml.Serialization
Imports System.Xml
Imports System.IO

Namespace XmlTransformation

	Public Class XmlSerializerHelper
		Public Function Serialize(ByVal obj As Object) As String
			Dim serializer = New XmlSerializer(obj.GetType())

			Dim writerSettings = New XmlWriterSettings With {.OmitXmlDeclaration = True, .Indent = True}

			Dim emptyNameSpace = New XmlSerializerNamespaces()
			emptyNameSpace.Add(String.Empty, String.Empty)

			Dim stringWriter = New StringWriter()
			Using xmlWriter = XmlWriter.Create(stringWriter, writerSettings)
				serializer.Serialize(xmlWriter, obj, emptyNameSpace)

				Return stringWriter.ToString()
			End Using
		End Function
	End Class
End Namespace